### DISCUSSION TITLE
Deploying multiple strategies

### DISCUSSION THREAD

**Discussion by Alpha23-Ltd** [nautechsystems/nautilus_trader#1058] [Q&A] [answered]
Hi, 

I have a market making Strategy that is running Binance Futures testnet without any issues, it references just one instrument and is run from a single node and .py file. The strategy config has `order_id_tag`="001"

When I duplicate this strategy (they are independent in terms of trading logic) and run it with a different node and a different .py file with `order_id_tag`="002", I get this error:

`2023-03-21T11:29:43.928165000Z [ERR] A23R-001.ExecClient-BINANCE: Error on handling b'{"stream":"UOYZIy2EhtebkAohWnbPHkRF1UzO7rOVJ2qyXsm0ySBxxzBuHhtmWVkSnVeBd2yt","data":{"e":"ORDER_TRADE_UPDATE","T":1679398183846,"E":1679398183850,"o":{"s":"BTCUSDT","c":"O-20230321-001-002-3","S":"BUY","o":"LIMIT","f":"GTX","q":"1","p":"27688.90","ap":"0","sp":"0","x":"CANCELED","X":"CANCELED","i":3305672212,"l":"0","z":"0","L":"0","n":"0","N":"USDT","T":1679398183846,"t":0,"b":"0","a":"27928.40000","m":false,"R":false,"wt":"CONTRACT_PRICE","ot":"LIMIT","ps":"BOTH","cp":false,"rp":"0","pP":false,"si":0,"ss":0}}}'
TypeError(an integer is required)
Traceback (most recent call last):
  File "/nautilus_trader/nautilus_trader/adapters/binance/futures/execution.py", line 248, in _handle_user_ws_message
    self._futures_user_ws_handlers[wrapper.data.e](raw)
  File "/nautilus_trader/nautilus_trader/adapters/binance/futures/execution.py", line 258, in _handle_order_trade_update
    order_update.data.o.handle_order_trade_update(self)
  File "/nautilus_trader/nautilus_trader/adapters/binance/futures/schemas/user.py", line 256, in handle_order_trade_update
    report = self.parse_to_order_status_report(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nautilus_trader/nautilus_trader/adapters/binance/futures/schemas/user.py", line 220, in parse_to_order_status_report
    return OrderStatusReport(
           ^^^^^^^^^^^^^^^^^^
  File "nautilus_trader/execution/reports.pyx", line 167, in nautilus_trader.execution.reports.OrderStatusReport.__init__`

The `OrderStatus` Report fails because `CONTRACT_PRICE` is being passed into TriggerType, which doesn't seem to be an enum that is supported.

`trigger_type=enum_parser.parse_binance_trigger_type(self.wt.value)`

I have found that i can get round this by adding my second strategy to the node of the first strategy and just deploying one .py file. Then the errors go away.

However, I guess this means bringing down the entire node if there is an issue with one strategy..
Is it more performant to run a node per instrument?

many thanks

**Comment by rsmb7z** [comment, marked as answer]
> However, I guess this means bringing down the entire node if there is an issue with one strategy..
> Is it more performant to run a node per instrument?

You can `stop` individual strategy if there is need and there is no need for seperate Node for each Instrument unless it is for some specfic purpose such as Observer etc.
